Idref.js

```diff
@@ -37,21 +37,31 @@
 
 var domain2translator = {
 	'www.sudoc.abes.fr': '1b9ed730-69c7-40b0-8a06-517a89a3a278',
-	'hal.archives-ouvertes.fr': '58ab2618-4a25-4b9b-83a7-80cd0259f896',
-	'edutice.archives-ouvertes.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
-	'halshs.archives-ouvertes.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
+	'hal.archives-ouvertes.fr': 'f20f91fe-d875-47e7-9656-0abb928be472',
 	'archivesic.ccsd.cnrs.fr': 'dedcae51-073c-48fb-85ce-2425e97f128d',
 	'memsic.ccsd.cnrs.fr': '58ab2618-4a25-4b9b-83a7-80cd0259f896',
 	'catalogue.bnf.fr': '47533cd7-ccaa-47a7-81bb-71c45e68a74d',
 	'www.theses.fr': '3f73f0aa-f91c-4192-b0d5-907312876cb9',
 	'www.persee.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
 	'oatao.univ-toulouse.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
 	'pub.orcid.org': 'bc03b4fe-436d-4a1f-ba59-de4d2d7a63f7',
-	'tel.archives-ouvertes.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
-	'medihal.archives-ouvertes.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
 	'dumas.ccsd.cnrs.fr': '951c027d-74ac-47d4-a107-9c3069ab7b48',
 };
 
+function getTranslatorFromDomain(domain) {
+	if (domain2translator[domain]) {
+		return domain2translator[domain];
+	}
+
+	/* Fallback for *.archives-ouvertes.fr */
+	if (domain.includes(".archives-ouvertes.fr")) {
```

Comment:
Not that it really matters, but for precision it should be `.endsWith('.archives-ouvertes.fr')`.